alicloud/resource_alicloud_logtail_config.go

```diff
@@ -0,0 +1,235 @@
+package alicloud
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/aliyun/aliyun-log-go-sdk"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+	"strings"
+	"time"
+)
+
+func resourceAlicoudLogtailConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicoudLogtaiConfiglCreate,
+		Read:   resourceAlicoudLogtailConfigRead,
+		Update: resourceAlicoudLogtaiConfiglUpdate,
+		Delete: resourceAlicoudLogtailConfigDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+
+			"config_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"input_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"log_sample": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"logstore": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"output_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Does the `output_type` have optional values?

Reply:
Currently not optional, optional fields have been added to facilitate future changes